core/gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -281,14 +287,33 @@ export class ResultPanelComponent implements OnInit, OnDestroy {
 
   resetPanelPosition() {
     this.dragPosition = { x: this.returnPosition.x, y: this.returnPosition.y };
+    this.isDocked = true;
+  }
+
+  checkPanelIsDocked() {
+    this.isDocked = this.returnPosition.x === this.dragPosition.x && this.returnPosition.y === this.dragPosition.y;
```

Comment:
Rename this function to `isPanelDocked`

since you have this condition to tell if panel is docked or not, you don't need `this.isDocked`. Whenever you want to check if the panel is docked or not, you call this function. This can reduce the complexity of maintaining another state variable.